java/src/main/java/ai/rapids/cudf/Table.java

```diff
@@ -757,6 +760,95 @@ public static Table concatenate(Table... tables) {
     return new Table(concatenate(tableHandles));
   }
 
+  /**
+   * Given a sorted table return the lower bound.
+   * Example:
+   *
+   *  Single column:
+   *      idx            0   1   2   3   4
+   *   inputTable  =   { 10, 20, 20, 30, 50 }
+   *   valuesTable =   { 20 }
+   *   result      =   { 1 }
+   *
+   *  Multi Column:
+   *      idx                0    1    2    3    4
+   *   inputTable      = {{  10,  20,  20,  20,  20 },
+   *                      { 5.0,  .5,  .5,  .7,  .7 },
+   *                      {  90,  77,  78,  61,  61 }}
+   *   valuesTable     = {{ 20 },
+   *                      { .7 },
+   *                      { 61 }}
+   *   result          = {  3 }
+   * NaNs in column values produce incorrect results.
+   * The input table and the values table need to be non-empty (row count > 0)
+   * The column data types of the tables' have to match in order.
+   * Strings and String categories do not work for this method. If the input table is
+   * unsorted the results are wrong. Types of columns can be of mixed data types.
+   * @param areNullsSmallest true if nulls are assumed smallest
+   * @param valueTable the table of values that need to be inserted
+   * @param descFlags indicates the ordering of the column(s), true if descending
+   * @return ColumnVector with lower bound indices for all rows in valueTable
+   */
+  public ColumnVector lowerBound(boolean areNullsSmallest,
+      Table valueTable, boolean[] descFlags) {
```

Comment:
Just curious what you think about using `OrderByArg... args` instead of `boolean[] descFlags`?  It would make the API much more consistent with `orderBy`, but to make it work we would have to rearrange the order of the columns being sent to the native code.  By this I mean `orderBy` can sort based off of any columns (i.e. sort on column 2 first, then column 9, then column 3) but `lowerBound` and `upperBound` need to have the columns ordered so the first key is the first column, second key is the second column etc.

Reply:
I have added a commit which incorporates orderByArg but does not change the order of the columns but instead makes a descFlags array based on the args. I have also added a test and I think this is better since the jni call remains closer to what cudf expects and the java calls remains closer to users might want. I hope I got the motivation of your comment addressed and if I have made a wrong assumption, feel free to point out. Thank u!

Reply:
That is not really what I was asking for.  I wanted to get your opinion if it would be a good change or not.  Not doing the column reordering makes if fundamentally incompatible with `orderBy`.  If I did the following

```
OrderByArg[] ordering = new OrderByArg[]{Table.asc(1)};
try (Table ordered = input.orderBy(true, ordering);
      Table orderedInsert =  insert.orderBy(true, ordering);
       ColumnVector locations = input.upperBound(true, insert, ordering)) {
  ...
}
```

it would produce inconsistent results because `orderBy` gives no guarantee about the ordering of column 0, but `upperBound` assumes that column 0 is in ascending order. Unless we do the column reordering we should just stick with the boolean array so it is obvious from the API the requirements placed on it.

Reply:
I understand. The API for bounds expects all the passed to columns to have been sorted based on the some order args. Sorry for having missed that. I think keeping it as boolean array makes most sense and the application using the API can send only a table which has  the (sorted) columns that were mentioned in the sort args. Will revert the last commit and update.